pkg/autodiscovery/listeners/kube_services.go

```diff
@@ -31,11 +32,12 @@ const (
 
 // KubeServiceListener listens to kubernetes service creation
 type KubeServiceListener struct {
-	informer   infov1.ServiceInformer
-	services   map[types.UID]Service
-	newService chan<- Service
-	delService chan<- Service
-	m          sync.RWMutex
+	informer      infov1.ServiceInformer
```

Comment:
Nit - given that all of these fields are not - does the struct itself need to be exported? `NewKubeServiceListener` returns `ServiceListener`. Just pointing out that it's impossible to initialize this outside of this package.

Reply:
yes I don't think it needs to be exported, though if we want to change this I think we should do it for all the listeners and providers for consistency - this could become out of scope for this PR which is already big 😅 
I'll backlog something to make sure we do this later, with other tasks we could do to improve the code quality of the AD package as well (improve the docs, the test coverage, update the readme files in the AD package which are very outdated btw)